include/seqan/arg_parse/arg_parse_version_check.h

```diff
@@ -0,0 +1,466 @@
+    std::string _path = std::string(getenv("UserProfile")) + "/.config/seqan";
+#else
+    std::string _path = std::string(getenv("HOME")) + "/.config/seqan";
+#endif
+    
+    //get system information
+#ifdef __linux
+    std::string _os = "Linux";
+#elif __APPLE__
+	std::string _os = "MacOS";
+#elif defined(PLATFORM_WINDOWS)
+	std::string _os = "Windows";
+#elif __FreeBSD__
+    std::string _os = "FreeBSD";
+#elif __OpenBSD__
+    std::string _os = "OpenBSD";
+#else
+    std::string _os = "unknown";
+#endif
+
+    // ----------------------------------------------------------------------------
+    // Constructors
+    // ----------------------------------------------------------------------------
+    VersionCheck(std::future<bool> & fut,
+                 std::string const & name,
+                 std::string const & version,
+                 std::string const & website):
+    _fut(fut)
+    {
+        _name = name;
+        if (!version.empty() &&
+            std::regex_match(version, std::regex("^[[:digit:]]\\.[[:digit:]]\\.[[:digit:]].*")))
+            _version = version.substr(0,5); // in case the git revision number is given take only version number
+        if(!website.empty())
+            _website = website;
+        _getProgram();
+        _updateCommand();
+    }
+
+    VersionCheck(VersionCheck const & rhs):
+    _fut(rhs._fut)
+    {
+        _name    = rhs._name;
+        _version = rhs._version;
+        _website = rhs._website;
+        _program = rhs._program;
+        _command = rhs._command;
+    }
+
+    VersionCheck(VersionCheck && rhs) :
+        _fut(rhs._fut)
+    {
+        swap(rhs);
+    }
+
+    VersionCheck & operator=(VersionCheck rhs)
+    {
+        swap(rhs);
+        return *this;
+    }
+
+    // ----------------------------------------------------------------------------
+    // Member Functions
+    // ----------------------------------------------------------------------------
+    inline void swap(VersionCheck & rhs)
+    {
+        std::swap(_name,    rhs._name);
+        std::swap(_version, rhs._version);
+        std::swap(_website, rhs._website);
+        std::swap(_program, rhs._program);
+        std::swap(_command, rhs._command);
+        std::swap(_fut,     rhs._fut);
+    }
+
+#ifdef __unix
+    void _getProgram()
+    {
+        // ask if system call for version or help is successfull
+        if (!system("wget --version > /dev/null 2>&1"))
+            _program = "wget -q -O";
+        else if (!system("curl --version > /dev/null 2>&1"))
+            _program =  "curl -o";
+#ifndef __linux
+        // ftp call does not work on linux
+        else if (!system("which ftp > /dev/null 2>&1"))
+            _program =  "ftp -Vo";
+#endif
+        else
+            _program.clear();
+    }
+#else // windows
+    void _getProgram()
+    {
+        _program = "powershell.exe -NoLogo -NonInteractive -Command \"& {Invoke-WebRequest -erroraction 'silentlycontinue' -OutFile";
```

Comment:
According to this, https://technet.microsoft.com/library/hh849901
Invoke-WebRequest has been supported since 3.0.

According to this, https://en.wikipedia.org/wiki/Windows_PowerShell 
Powercell 3.0 has been integrated since Windows 8.
(But Windows 7 / Server 2008 users can use it with a service pack.

Just for your information.